### PR TITLE
Fix beer search

### PR DESCRIPTION
**Pull request by mattbgold**

### CODE REVIEW

File: src/client/components/shared/typeahead/list/component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    this.cycleItem(1);
+                    break;
+                case 13: // enter
+                    event.preventDefault();
+                    if (this.highlightedItem >= 0) {
+                        this.selectElement(this.elements[this.highlightedItem]);
+                    }
+                    break;
+                case 27: // escape
+                    this.visible = false;
```
[the remark this answers is not in the document]
Comment (mattbgold) [reply src/client/components/shared/typeahead/list/component.ts]: Don't think so, escape key doesn't do anything weird.

File: src/server/services/mysql_database.ts [hunk 2]
```diff
@@ -241,6 +241,16 @@ export class MysqlDatabase implements Database {
         return this.query(q, params);
     }
 
+    searchForBeer(searchTerm: string): Observable<Beer[]> {
+        let q = 'Select * from `beers`'
+        + ' join `styles` on `styles`.`StyleId` = `beers`.`StyleId`'
+        + ' join `breweries` on `breweries`.`BreweryId` = `beers`.`BreweryId`'
+        + ' where `beerName` LIKE ? OR `breweryName` LIKE ? LIMIT 100';
+
+        return this.query(q, [`%${searchTerm}%`, `%${searchTerm}%`])
```
[the remark this answers is not in the document]
Comment (mattbgold) [reply src/server/services/mysql_database.ts]: good point, I'll probably have to strip out special characters here. will do some testing on it. really need some ORM

File: src/server/routes/admin.ts [hunk 3]
```diff
@@ -40,9 +42,29 @@ module.exports = (APP_CONFIG) => {
         .subscribe(tapInfo => sockets.emit('TapInfoEvent', tapInfo));
     }
 
+    function rankSearchResult(term: string, result: Beer) {
+        let regex =  new RegExp(term.split(' ').join('|'), 'gi');
```
Comment (justanr) [inline src/server/routes/admin.ts]: Should we move this down to the actual method where it's used to not recreate the regex every time? Or does Node do regex interning?

**Comment by justanr** [comment]
:+1: Just a small question about the regex otherwise looks good to me.

**Pull request merged**